include/usb/class/usb_audio.h

```diff
@@ -0,0 +1,286 @@
+					       struct net_buf *buffer,
+					       size_t size);
+
+/**
+ * @brief Callback type used to inform the app that Host has changed
+ *	  one of the features configured for the device.
+ *	  Applicable for all devices.
+ *
+ * @warning Host may not use all of configured features.
+ *
+ * @param evt Event information to be parsed by app.
+ */
+typedef void (*usb_audio_feature_updated_cb_t)(
+					struct usb_audio_feature_unit_evt evt);
+
+/**
+ * @brief Audio callbacks used to interact with audio devices by user App.
+ *
+ * usb_audio_ops structure contains all relevant callbacks to interact with
+ * USB audio devices. Each of this callbacks is optional and may be left NULL.
+ * This will not break the stack but could make USB audio device useless.
+ * Depending on the device some of those callbacks are necessary to make USB
+ * device work as expected sending/receiving data. For more information refer
+ * to callback documentation above.
+ */
+struct usb_audio_ops {
+	/* Callback called when data could be send */
+	usb_audio_data_request_cb_t data_request_cb;
+
+	/* Callback called when data were successfully written with sending
+	 * capable device. Applicable for headset and microphone. Unused for
+	 * headphones.
+	 */
+	usb_audio_data_completion_cb_t data_written_cb;
+
+	/* Callback called when data were successfully received by receive
+	 * capable device. Applicable for headset and headphones. Unused for
+	 * microphone.
+	 */
+	usb_audio_data_completion_cb_t data_received_cb;
+
+	/* Callback called when features were manipulation by Host */
+	usb_audio_feature_updated_cb_t feature_update_cb;
+};
+
+/** USB Audio device controls
+ *
+ * This structure present the structure of possible features that USB audio
+ * device may support. Enabling features is done via DT overlay.
+ */
+struct usb_audio_features {
+	bool  mute;
+	u16_t volume;
+	u8_t  tone_control[3];
+	/* Left for time being as u8_t
+	 * check Table 5-27 audio10.pdf
+	 */
+	u8_t  graphic_equalizer;
+	bool  automatic_gain_control;
+	u16_t delay;
+	bool  bass_boost;
+	bool  loudness;
+};
+
+/** @brief Get the frame size that is accepted by the host.
+ *
+ * This function returns the frame size for Input Devices that is expected
+ * by the Host. Returned value rely on Input Device configuration:
+ * - number of channels
+ * - sampling frequency
+ * - sample resolution
+ * Device configuration is done via DT overlay.
+ *
+ * @param dev The Input device that is asked for frame size.
+ *
+ * @warning Do not use with OUT only devices (Headphones).
+ *	    For OUT only devices this function shall return 0.
+ */
+size_t usb_audio_get_in_frame_size(const struct device *dev);
+
+/**
+ * @brief Register the USB Audio device and make it useable.
+ *	  This must be called in order to make the device work
+ *	  and respond to all relevant requests.
+ *
+ * @param dev USB audio device which will send the data
+ *	      over its ISO IN endpoint
```

Comment:
And what about OUT only devices?

Reply:
I must have forgotten to update it, its copied from ```usb_audio_send```..